go/saltpack/decrypt.go

```diff
@@ -249,13 +261,16 @@ func (ds *decryptStream) processEncryptionHeader(hdr *EncryptionHeader) error {
 		if longLivedSenderKey == nil {
 			return ErrNoSenderKey
 		}
-		ds.tagKey = secretKey.Precompute(longLivedSenderKey)
 		ds.mki.SenderKey = longLivedSenderKey
 	} else {
 		ds.mki.SenderIsAnon = true
 		ds.mki.SenderKey = ephemeralKey
 	}
 
+	// Compute the MAC key.
+	macKeyBox := secretKey.Box(ds.mki.SenderKey, nonceForMACKeyBox(ds.headerHash[:]), make([]byte, 32))
+	ds.macKey = macKeyBox[16:48]
```

Comment:
Can you use the constant in `nacl` for this `16`?  I think it's called `Overhead`.  Is it a crazy idea to just take the whole encryption btw?